lib/ansible/modules/cloud/azure/azure_rm_gallery_info.py

```diff
@@ -0,0 +1,270 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2019 Liu Qingyi, (@smile37773)
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_gallery_info
+version_added: '2.9'
+short_description: Get Gallery info.
+description:
+  - Get info of Gallery.
+options:
+  resource_group:
+    description:
+      - The name of the resource group.
+    type: str
+  name:
+    description:
+      - Resource name
+    type: str
+extends_documentation_fragment:
+  - azure
+author:
+  - Liu Qingyi (@smile37773)
+
+'''
+
+EXAMPLES = '''
+- name: List galleries in a subscription.
+  azure_rm_gallery_info:
+- name: List galleries in a resource group.
+  azure_rm_gallery_info:
+    resource_group: myResourceGroup
+- name: Get a gallery.
+  azure_rm_gallery_info:
+    resource_group: myResourceGroup
+    name: myGallery
+
+'''
+
+RETURN = '''
+galleries:
+  description: >-
+    A list of dict results where the key is the name of the Gallery and the
+    values are the facts for that Gallery.
+  returned: always
+  type: complex
+  contains:
+    id:
+      description:
+        - Resource Id
+      returned: always
+      type: str
+      sample: "/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+      /resourceGroups/myResourceGroup/providers/Microsoft.Compute/galleries/myGallery"
+    name:
+      description:
+        - Resource name
+      returned: always
+      type: str
+      sample: "myGallery"
+    type:
+      description:
+        - Resource type
+      returned: always
+      type: str
+      sample: "Microsoft.Compute/galleries"
+    location:
+      description:
+        - Resource location
+      returned: always
+      type: str
+      sample: "eastus"
+    tags:
+      description:
+        - Resource tags
+      returned: always
+      type: dict
+      sample: { "tag": "value" }
+    properties:
```

Comment:
properties should be flattened. so for instance value returned by azure_rm_gallery_info should be similar to structure of parameters passed to azure_rm_gallery.

Reply:
you should implement and use format_item() function to do that